Mathlib/Combinatorics/SimpleGraph/Clique.lean

```diff
@@ -655,4 +651,211 @@ theorem cliqueFinset_map_of_equiv (e : α ≃ β) (n : ℕ) :
 
 end CliqueFinset
 
+/-! ### Independent Sets -/
+
+section IndepSet
+
+variable {s : Set α}
+
+/-- An independent set in a graph is a set of vertices that are pairwise not adjacent. -/
+abbrev IsIndepSet (s : Set α) : Prop :=
+  s.Pairwise (fun v w ↦ ¬G.Adj v w)
+
+theorem isIndepSet_iff : G.IsIndepSet s ↔ s.Pairwise (fun v w ↦ ¬G.Adj v w) :=
+  Iff.rfl
+
+/-- An independent set is a clique in the complement graph and vice versa. -/
+@[simp] theorem isClique_compl : Gᶜ.IsClique s ↔ G.IsIndepSet s := by
+  rw [isIndepSet_iff, isClique_iff]; repeat rw [Set.Pairwise]
+  simp_all [compl_adj]
+
+/-- An independent set in the complement graph is a clique and vice versa. -/
+@[simp] theorem isIndepSet_compl : Gᶜ.IsIndepSet s ↔ G.IsClique s := by
+  rw [isIndepSet_iff, isClique_iff]; repeat rw [Set.Pairwise]
+  simp_all [compl_adj]
+
+instance [DecidableEq α] [DecidableRel G.Adj] {s : Finset α} : Decidable (G.IsIndepSet s) :=
+  decidable_of_iff' _ G.isIndepSet_iff
+
+/-- If `s` is an independent set, its complement meets every edge of `G`. -/
+lemma IsIndepSet.nonempty_mem_compl_mem_edge
+    [Fintype α] [DecidableEq α] {s : Finset α} (indA : G.IsIndepSet s) {e} (he : e ∈ G.edgeSet) :
+  { b ∈ sᶜ | b ∈ e }.Nonempty := by
+  obtain ⟨v , w⟩ := e
+  by_contra c
+  simp_all [filter_eq_empty_iff, isIndepSet_iff]
+  by_cases vins : v ∈ s
+  · have wins : w ∈ s := by by_contra wnins; exact (c wnins).right rfl
+    exact (indA vins wins (Adj.ne he)) he
+  · exact (c vins).left rfl
+
+/-- The neighbors of a vertex `v` form an independent set in a triangle free graph `G`. -/
+theorem isIndepSet_neighborSet_of_triangleFree [DecidableEq α] (h: G.CliqueFree 3) (v : α) :
+    G.IsIndepSet (G.neighborSet v) := by
+  by_contra nind
+  simp [SimpleGraph.IsIndepSet, Set.Pairwise] at nind
```

Comment:
We usually avoid non-terminal `simp` calls in mathlib: you can use `simp?` and follow the code action here. Similarly below

Reply:
No, those are definitely fine